server/channels/app/cloud.go

```diff
@@ -198,12 +198,17 @@ func (a *App) SendUpgradeConfirmationEmail(isYearly bool) *model.AppError {
 
 	embeddedFiles := make(map[string]io.Reader)
 	if isYearly {
-		pdf, filename, pdfErr := a.Cloud().GetInvoicePDF("", subscription.LastInvoice.ID)
-		if pdfErr != nil {
-			a.Log().Error("Error retrieving the invoice for subscription id", mlog.String("subscription", subscription.ID), mlog.Err(pdfErr))
+		lastInvoice := subscription.LastInvoice
+		if lastInvoice == nil {
+			a.Log().Error("Last invoice not defined for the subscription", mlog.String("subscription", subscription.ID))
```

Comment:
1/5 is this really an `Error`?

Could this case happen regularly if a fresh customer upgrades but didn't got an invoice yet?

Reply:
@nickmisasi @marianunez pinging the knowledgeable here :P Any idea in what case this could happen>

Reply:
This function would be after they upgraded and paid so they should have an invoice. If not, there was an error in some way. Now 0/5 if we want to log an error here or a warning given that it will still continue to send the upgrade confirmation email below

Reply:
Sorry for the delay, I didn't get notified for these mentions. 

Maria's right - at this point I'd expect the invoice to be non-zero given the current up-front nature of our product offering. I can't think of any reason it would be nil, so I think for now let's log it as an Error.

Some things I'll note:
I don't see any occurrences of this error in the last 30 days in production. Could this have come from someone's local instance or some test environment? 
We did previously have a bug where [this downgrade email would be sent to someone who downgraded to Free](https://mattermost.atlassian.net/browse/MM-48125?jql=text%20~%20%22upgrade%20email%22%20) in which case a nil lastInvoice _would_ be possible. That's since been fixed but definitely could have caused this

Reply:
@nickmisasi I was not granted a lot of info, just that we might have something panicking there and it's the only piece of code that was a candidate for a `nil` reference at this time. the worst-case scenario here is we added some protection around something that does not need it - it's not that bad haha. Thanks for the clarification tho, will ping my dear colleagues for re-review after your clarification.